src/ledger-keyring.ts

```diff
@@ -157,26 +156,25 @@ export class LedgerKeyring extends EventEmitter {
         };
       }
     }
-
+    const keys = new Set<string>(Object.keys(this.accountDetails));
     // try to migrate non-LedgerLive accounts too
     if (!this.#isLedgerLiveHdPath()) {
-      this.accounts
-        .filter(
-          (account) =>
-            !Object.keys(this.accountDetails).includes(
-              ethUtil.toChecksumAddress(account),
-            ),
-        )
-        .forEach((account) => {
-          try {
-            this.accountDetails[ethUtil.toChecksumAddress(account)] = {
+      this.accounts.forEach((account) => {
+        try {
+          const key = ethUtil.toChecksumAddress(account);
+
+          if (!keys.has(key)) {
+            this.accountDetails[key] = {
               bip44: false,
               hdPath: this.#pathFromAddress(account),
             };
-          } catch (error) {
-            console.log(`failed to migrate account ${account}`);
           }
-        });
+          // fix a edge case when accounts itself has duplicate address
```

Comment:
> It looks to me like this part is actually a fix unrelated to the performance improvement - does it warrant breaking out and a test-case? Also: Do we know that this is always fine?

actually i find that is no necessary, or it will not cause any issue if any edge case happen
e.g
accounts = [acc1, acc1, acc2] , accountDetails = {acc2}

if we take a dry run on the Prev code
1. filter accounts where those not in accountDetails, accounts become  [acc1, acc1]
2. acc1 not in accountDetails, acc1 add to accountDetails 
3. acc1 not in accountDetails, acc1 add to accountDetails , but the content will be same
4. at the end result will become accountDetails = {acc2, acc1}

if we take a dry run on the Updated code
1. acc2 in set, skip
2. acc1 not in set, acc1 add to accountDetails , add to set
3. acc1 in set, skip
4. at the end result will become accountDetails = {acc2, acc1}